docs/providers/aws/events/streams.md

```diff
@@ -69,7 +69,12 @@ functions:
       - stream:
           type: kinesis
           arn: arn:aws:kinesis:region:XXXXXX:stream/foobar
-          consumer: preExistingName
+          consumer: arn:aws:kinesis:us-west-2:xxx:stream/my-kinesis-stream/consumer/my-consumer
+      - stream:
+        type: kinesis
+        arn: arn:aws:kinesis:region:XXXXXX:stream/foobar
+        consumer: true
+        consumerName: 'my-consumer'
```

Comment:
I'm honestly not sure about having both `consumer` and `consumerName` properties as it's quite confusing. However, I have a hard time coming up with a better alternative because while object notation sounds interesting, it's always going to have only one property defined, e.g. if we allow such notation:

```
consumer: true
```
or 
```
consumer:
  name: ...
  arn: ... (only one of arn/name are allowed)
```

then we end up with situation where the nested config will only have a single property defined, either `arn` or `name` as having them both doesn't make sense. 

In current form, we should also have some prevention mechanism because if `consumer: true`, it doesn't really add any value because we can already derive that information from `consumerName` being defined. On the other hand, we should also validate that if `consumer: arn:..`, then `consumerName` is an invalid piece of configuration without any meaning. 

Do you have any other ideas @aheuermann on how the "perfect" configuration of consumer should look like?

@medikoo, maybe you have an opinion here?

Reply:
@pgrzesik I posted a comment in original issue: https://github.com/serverless/serverless/issues/9491#issuecomment-844860763 I think it's best if first over there we agree on solution and then move to PR